Sources/Purchasing/StoreKit2/StoreKit2ProductPurchaser.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  SK2ProductPurchaser.swift
+//
+//  Created by Will Taylor on 2/25/25.
+
+import Foundation
+import StoreKit
+
+internal protocol StoreKit2ProductPurchaserType {
+
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    func purchase(
+        product: PurchasableSK2Product,
+        options: Set<StoreKit.Product.PurchaseOption>,
+        storeKit2ConfirmInOptions: StoreKit2ConfirmInOptions?
+    ) async throws -> StoreKit.Product.PurchaseResult
+}
+
+internal class StoreKit2ProductPurchaser: StoreKit2ProductPurchaserType {
+
+    private let systemInfo: SystemInfo
+
+    init(
+        systemInfo: SystemInfo
+    ) {
+        self.systemInfo = systemInfo
+    }
+
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    func purchase(
+        product: PurchasableSK2Product,
+        options: Set<StoreKit.Product.PurchaseOption>,
+        storeKit2ConfirmInOptions: StoreKit2ConfirmInOptions?
+    ) async throws -> StoreKit.Product.PurchaseResult {
+#if VISION_OS
```

Comment:
Please please double check this section carefully - I've reread it multiple times, but this is still a risky change